### PR TITLE
feat: User-command "HotReload"

### PR DESCRIPTION
**Pull request by Jint-lzxy**
This commit modularizes the hot-reload feature. The user can execute hot-reload through `<leader>R` or `:HotReload`. 

@CharlesChiuGit Can you help me test whether the expected result is achieved?

### CODE REVIEW

**Comment by CharlesChiuGit** [comment]
wow, so fast!

**Comment by ayamir** [comment]
Make it a configurable setting at `settings.lua` and default equals `false` is more reasonable.

**Comment by CharlesChiuGit** [comment]
Maybe we can also revert the workaround for disable notify on `PackerCompile`?

File: lua/core/init.lua [hunk 1]
```diff
@@ -69,6 +69,13 @@ local disable_distribution_plugins = function()
 	-- vim.g.loaded_remote_plugins = 1
 end
 
+local set_hot_reload = function()
+	vim.api.nvim_create_user_command("HotReload", function()
+		require("core.pack").back_compile(true)
+		vim.api.nvim_command([[source $MYVIMRC | redraw! | echom "Hot-reload success!"]])
```
Comment (CharlesChiuGit) [inline lua/core/init.lua:75]: we also need to source the current changed file, so the changes will be valid.

Comment (Jint-lzxy) [reply lua/core/init.lua]: Then we would need to judge whether the currently saved file is legal? _(whether it is in the correct path and whether the extension is `*.lua`)_

Comment (Jint-lzxy) [reply lua/core/init.lua]: I don't think it has been resolved yet. `neovim` will also source extraneous files, we should check before execute

Comment (CharlesChiuGit) [reply lua/core/init.lua]: yeah, i think it will be necessary.

Comment (CharlesChiuGit) [reply lua/core/init.lua]: > @CharlesChiuGit I don't think it has been resolved yet. `neovim` will also source extraneous files, we should check before execute

Sorry, will do.

**Comment by Jint-lzxy** [comment]
I agree with @ayamir. Frequent reloads can cause some potential problems, and many plugins actually don't support hot reloading their settings when they are `required` again. Gonna close this now, what's your opinion? @CharlesChiuGit @ayamir

**Comment by ayamir** [comment]
> Gonna close this now

We need to revert corresponding commits.

**Comment by ayamir** [comment]
> Ok, sorry for causing troubles.

No need to say sorry, just be careful next time. :)

**Comment by Jint-lzxy** [comment]
> Ok, sorry for causing troubles.

Yeah. A very good feature, but not _that_ practical to implement it 👍

**Pull request closed without merging**